dts/bindings/sensor/plantower,pmsX003.yaml

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2020, Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Plantower PMSX003 air pollution sensor.
+  Air pollution sensor capable of measuring mass concentration of dust.
+  The 'X' in PMSX003 stands for any sensor in the series, such as PMS6003, PMS7003, or PMS5003.
+  See more info at:
+  https://www.espruino.com/datasheets/PMS7003.pdf
+  https://www.digikey.com/en/htmldatasheets/production/2903006/0/0/1/pms5003-series-manual
+  https://evelta.com/content/datasheets/203-PMS6003.pdf
+
+
+compatible: "plantower,pmsX003"
```

Comment:
doesn't seem reasonable to me to rename the binding in this case, it will break downstream boards, and according to DT spec it is completely valid to specify the pms7003 compatible if the driver _is compatible_ with the device

eventually want to have like what linux has with something like regex matching for compatible groups but that's not something for this PR to do

kind of related problem as with https://github.com/zephyrproject-rtos/zephyr/pull/73512 and https://github.com/zephyrproject-rtos/zephyr/pull/73136

Reply:
Got, your point. I also renamed the folder and file name from pms7003 to pmsX003. So, should I also rename them back to the original name? what do you suggest?